Adding a bong and its spawn conditions

SJTheNewOne opened the pull request:
<!-- HOW TO USE: Under each "#### Heading" below, enter information relevant to your pull request.
Leave the headings unless they don't apply to your PR.

Please read carefully.
Once a pull request is submitted, automatic stylistic and consistency checks will be performed on the PR's changes.
The results can be either seen under the "Files changed" section of a PR or in the check's details.

Rules for suggested pull requests:
- If possible, limit yourself to small changes, 500 strings at max. Exceptions are adding or changing maps, and changes, that won't work unless they are done in a single run (even then there can be ways) - violating it puts a lot of unnecessary work on our merge team.
- Do not scope creep. If you make a pull request "Add new gun", please do not make anything more than adding the gun and following changes, like changing the stats of the gun, removing other guns from itemgroups or tweaking zombie horse stats - violating it makes future search and debugging stuff much harder, since PR name is not related to what is changed in the game. "Who the hell removed the quest item from drop in location X in PR, that adds a new plushie" - this may be a quote from a person who was affected by scope creep
- Do not make omnibus PRs. Meaning do not make a single PR, that fixes ten different, not related issues, at once, even if they are all one string - same as previously mentioned scope creep, it doesn't help to search the changes when debugging, despite all power of git blame tool

NOTE: Please grant permission for repository maintainers to edit your PR.  It is EXTREMELY common for PRs to be held up due to trivial changes being requested and the author being unavailable to make them. -->

#### Summary
Content "Add water pipe and spawns"
<!-- This section should consist of exactly one line, edit the one above.
1. Replace the word "Category" with one of these words: Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N.
2. Replace the text inside the quotes with a brief description of your changes.
Or if you don't want a changelog entry, replace the whole line with just the word "None" (with no quotes).
Examples:
1. None
2. Features "In-game Armor sprite change"
3. Interface "Show crafting failure chances in the crafting interface"
4. Infrastructure "JSON-ize slot machines"
5. Bugfixes "Crafting GUI: show how much recipe makes for non-charge items"
For more on the meaning of each category, see:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md
If approved and merged, your summary will be added to the project changelog:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/changelog.txt -->


#### Purpose of change

Adding a water pipe, also known as a "bong" to the game - an item that can be used for smoking, and also can contain some water. This specific example is based off of mass-manufactured water pipes which are often found in headshops, sold under the pretense of being used for tobacco - even in legalized areas, you find bongs being sold with labels saying something to the effect of 'tobacco use only' due to manufacture often taking place in countries where cannabis is highly illegal.  

#### Describe the solution

Adds a tool with the capacity to hold 500 ml of liquid, be used for smoking, boiling, and other purposes. A merger between a 0.5 liter jar and a tobacco pipe, code-wise. 

#### Describe alternatives you've considered

Suffering through my stoner character using a dry pipe. Clearly, this breaks my immersion. 

#### Testing

I tested cooking (was able to boil water and make soup without crashes), containing (held items just fine), smoking, and spawning. It can be used without issue when full of liquid, replicating the actual process of smoking with a water pipe. 

#### Additional context

This was a modding experiment that got approval from the community and people asked if I would add it. I decided to give it a swing. 

![Screenshot 2024-12-04 111042](https://github.com/user-attachments/assets/7f9305c1-ad68-435a-a27c-4cdabc420f73)
![Screenshot 2024-12-04 120953](https://github.com/user-attachments/assets/22d45b70-fc95-4ec2-9510-4bb617e0b631)
![Screenshot 2024-12-04 124032](https://github.com/user-attachments/assets/b7d3f41e-bbaa-4478-8508-7deca0f9e0cb)
![Screenshot 2024-12-04 131804](https://github.com/user-attachments/assets/7e396c5b-5e99-4f90-b32b-4f47b50181df)
![Screenshot 2024-12-04 110946](https://github.com/user-attachments/assets/91ae3b3e-d171-4edc-a842-a3dc543da456)



<!--README: Cataclysm: Dark Days Ahead is released under the Creative Commons Attribution ShareAlike 3.0 license.
The code and content of the game are free to use, modify, and redistribute for any purpose whatsoever.
By contributing to the project you agree to the terms of the license and that any contribution you make will also be covered by the same license.
See http://creativecommons.org/licenses/by-sa/3.0/ for details. -->

SJTheNewOne committed:
Adding a bong and its spawn conditions

This adds a new item, the water pipe, which represents glass water pipes, also known as "Bongs" and spawn conditions for them.

SJTheNewOne committed:
Update data/json/itemgroups/stashes.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SJTheNewOne committed:
Update data/json/mapgen/headshop.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

DSeyka commented on data/json/items/tool/smoking.json:
```diff
@@ -58,6 +58,36 @@
+    "type": "TOOL",
+    "name": { "str": "water pipe" },
+    "looks_like": "bottle_plastic",
+    "description": "A mass-manufacured water-pipe, otherwise known as a 'bong.' Headshops often insist these are used solely for tobacco, despite their iconic status as cannabis paraphernalia.",
```
```suggestion
    "description": "A mass-manufactured water-pipe, otherwise known as a 'bong.'  Head shops often insist these are used solely for tobacco, despite their iconic status as cannabis paraphernalia.",
```

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: DSeyka <66876732+DSeyka@users.noreply.github.com>

SJTheNewOne committed:
Update smoking.json

SJTheNewOne committed:
Update smoking.json

RedMisao commented:
I wasn't aware this was not already present in game.
Also, are these fragile? I've seen videos where these things seem to be indestructible.

SJTheNewOne commented:
> I wasn't aware this was not already present in game. Also, are these fragile? I've seen videos where these things seem to be indestructible.

Yeah I was surprised too. As for indestructibility, I only really have my own anecdotes but I've had enough sheepish people walk up to me saying "yeah...I broke my bong...." that I cannot imagine it would last long being used as a bludgeon. Thick glass can only get you so far.

SJTheNewOne committed:
Update smoking.json

Added variants

SJTheNewOne committed:
Update misc.json

Upped spawn in 'stoner' group.

SJTheNewOne commented:
I adjusted the spawn value and added variants, simply for flavour purposes. 
![Screenshot 2024-12-04 203321](https://github.com/user-attachments/assets/26217b9a-bb3b-4780-a8dd-57f432a82f57)
![Screenshot 2024-12-04 212806](https://github.com/user-attachments/assets/255243a1-6c2a-4386-bc08-ad0b858e4923)
![Screenshot 2024-12-04 212827](https://github.com/user-attachments/assets/6fbd9ea7-a98d-4f00-8353-394ff5766d00)

DSeyka commented on data/json/items/tool/smoking.json:
```diff
@@ -58,6 +58,81 @@
+      {
+        "id": "pipe_water_normal",
+        "name": { "str": "bong" },
+        "description": "A mass-manufacured water-pipe, otherwise known as a 'bong.' This one is made of plain, clear, glass.",
```
```suggestion
        "description": "A mass-manufactured water pipe, otherwise known as 'bong.'  This one is made of plain, clear glass.",
```

DSeyka commented on data/json/items/tool/smoking.json:
```diff
@@ -58,6 +58,81 @@
+      {
+        "id": "pipe_water_cartoon",
+        "name": { "str": "cartoon bong" },
+        "description": "A mass-manufacured water-pipe, otherwise known as a 'bong.' It is covered in copyrighted cartoon characters, all smoking marijuana.",
```
```suggestion
        "description": "A mass-manufactured water pipe, otherwise known as 'bong.'  It is covered in copyrighted cartoon characters, all smoking marijuana.",
```

DSeyka commented on data/json/items/tool/smoking.json:
```diff
@@ -58,6 +58,81 @@
+      {
+        "id": "pipe_water_rasta",
+        "name": { "str": "rasta bong" },
+        "description": "A mass-manufacured water-pipe, otherwise known as a 'bong.' It colored with the 'rasta' pattern of red, yellow, and green.",
```
```suggestion
        "description": "A mass-manufactured water pipe, otherwise known as 'bong.'  It is colored with the 'rasta' pattern of red, yellow, and green.",
```

DSeyka commented on data/json/items/tool/smoking.json:
```diff
@@ -58,6 +58,81 @@
+      {
+        "id": "pipe_water_handblown",
+        "name": { "str": "handblown bong" },
+        "description": "A handblown water pipe, also known as as a 'bong.' Lovely crafted by an artisan, a true fusion of aesthetic sensibilities and stoner functionalities ",
```
```suggestion
        "description": "A handblown water pipe, also known as as 'bong.'  Lovely crafted by an artisan, a true fusion of aesthetic sensibilities and stoner functionalities.",
```

Maleclypse replied:
```suggestion
        "description": "A handblown water pipe, also known as a 'bong.' Lovingly crafted by an artisan, a true fusion of aesthetic sensibilities and stoner functionalities ",
```

DSeyka commented on data/json/items/tool/smoking.json:
```diff
@@ -58,6 +58,81 @@
+      {
+        "id": "pipe_water_psychedelic",
+        "name": { "str": "psychedelic bong" },
+        "description": "A mass-manufactured water pipe, also known as a 'bong.' It is covered in stickers depicting 'psychedelic' art.",
```
```suggestion
        "description": "A mass-manufactured water pipe, also known as 'bong.'  It is covered in stickers depicting 'psychodelic' art.",
```

DSeyka commented on data/json/items/tool/smoking.json:
```diff
@@ -58,6 +58,81 @@
+      {
+        "id": "pipe_water_frosted",
+        "name": { "str": "frosted bong" },
+        "description": "A mass-manufactured water pipe, also known as a 'bong.' It is made out of frosted glass.",
```
```suggestion
        "description": "A mass-manufactured water pipe, also known as 'bong.'  It is made out of frosted glass.",
```

DSeyka commented on data/json/items/tool/smoking.json:
```diff
@@ -58,6 +58,81 @@
+      {
+        "id": "pipe_water_fakegold",
+        "name": { "str": "gold bong" },
+        "description": "A mass-manufactured water pipe, also known as a 'bong.' Though it looks like it's made out of gold, that is in fact just metallic paint.",
```
```suggestion
        "description": "A mass-manufactured water pipe, also known as 'bong.'  Though it looks like it's made out of gold, that is, in fact, just metallic paint.",
```

SJTheNewOne replied:
"a 'bong' " is the correct way of writing it. "Known as 'bong' " makes it sound like a proper noun, which it is not.

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: Maleclypse <54345792+Maleclypse@users.noreply.github.com>

SJTheNewOne committed:
Update smoking.json

SJTheNewOne committed:
Update smoking.json

SJTheNewOne committed:
Update smoking.json

Night-Pryanik commented on data/json/items/tool/smoking.json:
```diff
@@ -58,6 +58,81 @@
+    "type": "TOOL",
+    "name": { "str": "water pipe" },
+    "looks_like": "bottle_plastic",
+    "description": "A mass manufactured water pipe, otherwise known as a 'bong.' Head shops often insist these are used solely for tobacco, despite their iconic status as cannabis paraphernalia.",
```
```suggestion
    "description": "A mass-manufactured water pipe, otherwise known as a 'bong'.  Head shops often insist these are used solely for tobacco, despite their iconic status as cannabis paraphernalia.",
```

Night-Pryanik commented on data/json/items/tool/smoking.json:
```diff
@@ -58,6 +58,81 @@
+      {
+        "id": "pipe_water_normal",
+        "name": { "str": "bong" },
+        "description": "A mass manufactured water pipe, otherwise known as a 'bong.' This one is made of plain, clear, glass.",
```
```suggestion
        "description": "A mass-manufactured water pipe, otherwise known as a 'bong'.  This one is made of plain, clear, glass.",
```

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: Anton Burmistrov <Night_Pryanik@mail.ru>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: Anton Burmistrov <Night_Pryanik@mail.ru>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: Anton Burmistrov <Night_Pryanik@mail.ru>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: Anton Burmistrov <Night_Pryanik@mail.ru>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: Anton Burmistrov <Night_Pryanik@mail.ru>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: Anton Burmistrov <Night_Pryanik@mail.ru>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: Anton Burmistrov <Night_Pryanik@mail.ru>

SJTheNewOne committed:
Update data/json/items/tool/smoking.json

Co-authored-by: Anton Burmistrov <Night_Pryanik@mail.ru>

SJTheNewOne committed:
Update smoking.json

changing to-hit

SJTheNewOne commented on data/json/items/tool/smoking.json:
```diff
@@ -58,6 +58,81 @@
+        "moves": 400
+      }
+    ],
+    "to_hit": -2,
```
Done. During resolving some other issue, I think I uploaded an older version by accident. Apologies.

Karol1223 commented on data/json/items/tool/smoking.json:
```diff
@@ -58,6 +58,81 @@
+    "volume": "900 ml",
+    "longest_side": "30 cm",
+    "flags": [ "FRAGILE_MELEE", "NONCONDUCTIVE" ],
+    "price": 2000,
+    "price_postapoc": 100,
```
You need to use proper price syntax

SJTheNewOne replied:
The change has been made, I think.

SJTheNewOne committed:
Update smoking.json

Karol1223 reviewed (approved):
Seems fine to me so far. I'd also ask for a disassembly to go in hand with other simple glass items (so blind easy, 1 second, throwing it on the ground to shatter it to shards) but it's not technically necessary to go through.

Maleclypse commented:
> Seems fine to me so far. I'd also ask for a disassembly to go in hand with other simple glass items (so blind easy, 1 second, throwing it on the ground to shatter it to shards) but it's not technically necessary to go through.

I figure this can be a followup PR as it looks like this person was a new contributor and had to request test approvals each time they updated the PR.
